### PR TITLE
various updates

### LINKED ISSUE #18: Listing shouldn't require Github description
Right now to list a package we require them to have a Github repo description.

This requirement looks unnecessary and should be removed.

### PR DESCRIPTION
**Pull request by eddiemonge**
* Clean up some code styling
 * More logging
 * Change update interval to be one second beyond an hour
 * Updated response to be slimmer and what is being used for site
 * Hopefully better promise handling and fetching
 * Uses longer of GH or NPM description
 * Dont choke if GH doesn't respond
 * Dont require GH response

closes #18

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,130 +1,128 @@
...
+'use strict';
+
 var Q = require('q');
 var npmKeyword = require('npm-keyword');
 var packageJson = require('package-json');
 var ghGot = require('gh-got');
 var EventEmitter = require('events').EventEmitter;
 
-var UPDATE_INTERVAL_IN_SECONDS = 60 * 60;
+var UPDATE_INTERVAL_IN_SECONDS = 60 * 60 + 1;
```
Comment (SBoudrias) [inline index.js]: Why?

File: index.js [hunk 2]
```diff
@@ -1,130 +1,128 @@
...
+    owner: {
+      login: '',
+      html_url: '',
+    },
+    html_url: '',
+    stargazers_count: '',
+    watchers_count: ''
+  };
+
+  var description = npm.description && gh.description ? (
```
Comment (SBoudrias) [inline index.js]: Can we make this legible? It's very hard to see what is going on here. Use `if/else` when it's becoming very complex.

File: index.js [hunk 3]
```diff
@@ -1,130 +1,128 @@
...
     }
+    promise.resolve(transform ? transform(data) : data);
+  };
+}
 
-    d.resolve(packages);
-  });
-
+function getNpmPackageList() {
+  console.log('Querying for keywords');
```
Comment (SBoudrias) [inline index.js]: Let's add data about what is being query-ed.

File: index.js [hunk 4]
```diff
@@ -1,130 +1,128 @@
...
+  var d = Q.defer();
+  npmKeyword(
+    process.env.NPM_LIST_KEYWORD || 'yeoman-generator',
+    promiseCallback(d)
+  );
   return d.promise;
 }
 
 function getNpmPackages(list) {
+  console.log('Getting ' + list.length + ' packages');
```
Comment (SBoudrias) [inline index.js]: s/Getting/Received ?

**Comment by SBoudrias** [comment]
Are we okay with just `console.log` or should we use a logger that support threads?

File: index.js (hunk 1 above)
Comment (sindresorhus) [inline index.js]: I assume it's padded to make sure we're on a new rate limit hour. I would make that `1` => `10` just to be sure.

Comment (eddiemonge) [inline index.js]: yeah padding. I can change it to `+ 10`

Comment (sindresorhus) [inline index.js]: Make it a constant and use that. That way you don't need a comment to describe it.

File: index.js (hunk 4 above)
Comment (eddiemonge) [inline index.js]: its saying whats its going to be doing instead of what its starting with

**Comment by SBoudrias** [comment]
@eddiemonge why did you decide to close this PR?

**Comment by eddiemonge** [comment]
just you wait. if this new way is worse, I'll come back to this. But ... just ... you ... wait ... and ... see ...

**Pull request closed without merging**